src/app/services/skill-service.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Skill } from '../models/skill';
+import { Observable } from 'rxjs';
+
+
+@Injectable()
+export class SkillService {
+
+private skillsUrl: string;
+private saveSkillUrl: string;
+private deleteSkillUrl: string;
+private relinkSkillsUrl: string;
+
+private  returnval : Observable<Skill[]>;
+private uskills: Skill[];
+private    pskills: Observable<Skill[]> ;
+
+  constructor(private http: HttpClient) {
+    this.skillsUrl = 'http://localhost:8080/getskills';
+    this.saveSkillUrl = 'http://localhost:8080/saveskill';
+    this.deleteSkillUrl = 'http://localhost:8080/deleteskill';
+    this.relinkSkillsUrl = 'http://localhost:8080/relinkskills';
+  }
+
+  public findAll(): Observable<Skill[]> {
+    return this.http.get<Skill[]>(this.skillsUrl);
+  }
+
+  public deleteSkill(skill: Skill) {
```

Comment:
If all functions return an Observable, please add that to method as you've done on line 26.

Reply:
adding to both front end and backend: error handling. Exception should be caught in backend, backend should return the error message in the DTO object.

Reply:
will add this error handling after merge of feature NEX-154 implementing standard CRUD

Reply:
Still no return type added to methods on lines 30 / 35 / 40. Also don't forget to add comments to any method in this pull request.

Reply:
aangepast